### ISSUE TITLE
Implement keyboard focus (enhancement)

### ISSUE THREAD

**Issue by Briseur** [unarix/haiku_darkstyle#12]
The suggestion was previously made on Haiku forum in [this thread](https://discuss.haiku-os.org/t/flat-decorator-and-controllook/9611). Here is a more detailed version.

Actually keyboard navigation is difficult with Flat ControlLook because it misses keyboard focus visuals.
See [here](https://discuss.haiku-os.org/t/flat-decorator-and-controllook/9611/164?u=starcrasher) for comparison with other ControlLook.

I tried different mock ups to see what could be done.

First, I tried to see if other CL solutions could fit with Flat

- Add a border of navigation colour to the button with keyboard focus like does default Haiku CL.
  Flat CL buttons are slightly rounded so it would be difficult/impossible to implement or won't look good.

- Underline the text of the button with keyboard focus of navigation colour as the Be CL does.
  Use of navigation colour makes it more consistent but even with all buttons of the same size, IMHO it breaks the neat style. 
![suggestion_b_select](https://user-images.githubusercontent.com/37412120/137617843-2a6b279b-5bf4-40f5-9ac7-3e1e95ac42a3.png)

As it wasn't convincing enough, I tried to explore other solutions.

- Keep bigger button for the default button and but we are using colour for keyboard focus.
  It looks good but this is not consistent with what other CL are doing and it's somehow a bit confusing.
![Expect_select](https://user-images.githubusercontent.com/37412120/137617610-9a913656-4b19-4705-876f-f752a3994875.png)

- Add a dot of navigation colour as I suggested on the forum [here](https://discuss.haiku-os.org/t/flat-decorator-and-controllook/9611/166?u=starcrasher). 
 It is consistent and doesn't break the style. 
![suggestion_select](https://user-images.githubusercontent.com/37412120/137619903-71786c09-135e-4a56-87cb-2fb75ae06968.png)

- Same one but with all buttons all of the same size. Would it ease dot positioning?
  IMHO, this even looks better.
![suggestion_a_select](https://user-images.githubusercontent.com/37412120/137620024-3f6e5c2b-b40e-4977-b2f8-c4e6c0d914af.png)

**Comment by Briseur** [comment]
Keyboard focus is lacking for buttons, but also for list menus (i.e. Appearance colour list) and drop down menus (i.e. Appearance Decorator & ControlLook menus).
In case of list menus, I suggest to use the some border of navigation colour as in other CL.
For drop down menu widgets, since they are slightly rounded, I suggest to use a dot of navigation colours in bottom right like for buttons.
I may have miss something but in other places, tabs, text fields, boxes, etc the keyboard focus seems working.

**Comment by Briseur** [comment]
So I made new mock ups to illustrate previous post.


This is what I suggest for list menus. Exactly what is done in other CL.
![Flat_list](https://user-images.githubusercontent.com/37412120/138062900-48deeddb-7252-4a90-b1e5-96bed4f0665f.png)


This is what I had in mind for drop down menus, consistent with what I suggested already for buttons.
![Flat_dropdown](https://user-images.githubusercontent.com/37412120/138062963-c5189b9b-c117-4866-88b0-776612524d7e.png)

This is an alternative suggestion for drop down menus.
![Flat_dropdown_2](https://user-images.githubusercontent.com/37412120/138062982-507664ed-3c18-42e8-90bd-8b0e20944706.png)

I tried to move the dot above or under the arrow but, it wasn't looking good. So, I put the colour on the arrow itself.
![Flat_dropdown_3](https://user-images.githubusercontent.com/37412120/138068723-985b234a-70f4-4d13-87bc-5be374d19826.png)

I hope that these mock ups would help to implement keyboard focus where it is lacking. Though as l'm not a developer myself, I have no idea if some of the suggestions are feasible or not. 

On a side note, I noticed that in Qt apps, default button is (sometimes?) loosing its border. Perhaps having all buttons of same size would help with that...
